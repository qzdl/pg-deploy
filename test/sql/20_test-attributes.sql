--              ██     ██          ██ ██                ██
--             ░██    ░██         ░░ ░██               ░██
--   ██████   ██████ ██████ ██████ ██░██      ██   ██ ██████  █████   ██████
--  ░░░░░░██ ░░░██░ ░░░██░ ░░██░░█░██░██████ ░██  ░██░░░██░  ██░░░██ ██░░░░
--   ███████   ░██    ░██   ░██ ░ ░██░██░░░██░██  ░██  ░██  ░███████░░█████
--  ██░░░░██   ░██    ░██   ░██   ░██░██  ░██░██  ░██  ░██  ░██░░░░  ░░░░░██
-- ░░████████  ░░██   ░░██ ░███   ░██░██████ ░░██████  ░░██ ░░██████ ██████
--  ░░░░░░░░    ░░     ░░  ░░░    ░░ ░░░░░    ░░░░░░    ░░   ░░░░░░ ░░░░░░
-- attributes:
--
BEGIN;
CREATE SCHEMA if NOT EXISTS testr;
CREATE SCHEMA if NOT EXISTS testp;
DROP TABLE if EXISTS testr.a;
DROP TABLE if EXISTS testp.a;

CREATE TABLE testr.a(i int, ii text, iii bit);
CREATE TABLE testp.a(ii text, iv numeric CONSTRAINT positive_price CHECK (iv > 0));


SELECT pgdeploy.reconcile_table_attributes('testr', 'testp', 'a', 'a');

END;
