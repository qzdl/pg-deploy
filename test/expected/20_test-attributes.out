--              ██     ██          ██ ██                ██
--             ░██    ░██         ░░ ░██               ░██
--   ██████   ██████ ██████ ██████ ██░██      ██   ██ ██████  █████   ██████
--  ░░░░░░██ ░░░██░ ░░░██░ ░░██░░█░██░██████ ░██  ░██░░░██░  ██░░░██ ██░░░░
--   ███████   ░██    ░██   ░██ ░ ░██░██░░░██░██  ░██  ░██  ░███████░░█████
--  ██░░░░██   ░██    ░██   ░██   ░██░██  ░██░██  ░██  ░██  ░██░░░░  ░░░░░██
-- ░░████████  ░░██   ░░██ ░███   ░██░██████ ░░██████  ░░██ ░░██████ ██████
--  ░░░░░░░░    ░░     ░░  ░░░    ░░ ░░░░░    ░░░░░░    ░░   ░░░░░░ ░░░░░░
-- attributes:
--
BEGIN;
SET client_min_messages TO WARNING;
CREATE EXTENSION pgdeploy;
CREATE SCHEMA testr;
CREATE SCHEMA testp;
CREATE TABLE testr.a(i int, ii text, iii bit);
CREATE TABLE testp.a(ii text, iv numeric CONSTRAINT positive_price CHECK (iv > 0));
SELECT pgdeploy.reconcile_table_attributes(
    'testr'::name, 'a'::name, 'testr.a'::regclass::oid
    ,'testp'::name, 'a'::name, 'testp.a'::regclass::oid);
           reconcile_table_attributes            
-------------------------------------------------
 ALTER TABLE testr.a ADD COLUMN iv numeric NULL;
 ALTER TABLE testr.a DROP COLUMN i;
 ALTER TABLE testr.a DROP COLUMN iii;
 -- COLUMN: no change for ii
(4 rows)

-- CLEAN UP
DROP EXTENSION pgdeploy CASCADE;
ROLLBACK;
